impl_core/src/main/java/io/opencensus/trace/export/SampledSpanStoreImpl.java

```diff
@@ -0,0 +1,273 @@
+          + NUM_SAMPLES_PER_ERROR_BUCKET * NUM_ERROR_BUCKETS;
+  private final Map<String, PerSpanNameSamples> samples;
+
+  private static final class Bucket {
+
+    private final EvictingQueue<SpanImpl> queue;
+    private long nextSampleTime;
+
+    private Bucket(int numSamples) {
+      queue = EvictingQueue.create(numSamples);
+    }
+
+    private void considerForSampling(SpanImpl span) {
+      long spanEndTime = span.getEndNanoTime();
+      if (spanEndTime > nextSampleTime) {
+        queue.add(span);
+        nextSampleTime = spanEndTime + TIME_BETWEEN_SAMPLES;
+      }
+    }
+
+    private void getSamples(int maxSpansToReturn, List<SpanImpl> output) {
+      for (SpanImpl span : queue) {
+        if (output.size() >= maxSpansToReturn) {
+          break;
+        }
+        output.add(span);
+      }
+    }
+
+    private void getSampledFilteredByLatency(
+        long latencyLowerNs, long latencyUpperNs, int maxSpansToReturn, List<SpanImpl> output) {
+      for (SpanImpl span : queue) {
+        if (output.size() >= maxSpansToReturn) {
+          break;
+        }
+        long spanLatencyNs = span.getLatencyNs();
+        if (spanLatencyNs >= latencyLowerNs && spanLatencyNs < latencyUpperNs) {
+          output.add(span);
+        }
+      }
+    }
+
+    private int getNumSamples() {
+      return queue.size();
+    }
+  }
+
+  /**
+   * Keeps samples for a given span name. Samples for all the latency bucketes and one special
+   * bucket for errors.
+   */
+  private static final class PerSpanNameSamples {
+
+    private final Bucket[] latencyBuckets;
+    private final Bucket[] errorBuckets;
+
+    private PerSpanNameSamples() {
+      latencyBuckets = new Bucket[NUM_LATENCY_BUCKETS];
+      for (int i = 0; i < NUM_LATENCY_BUCKETS; i++) {
+        latencyBuckets[i] = new Bucket(NUM_SAMPLES_PER_LATENCY_BUCKET);
+      }
+      errorBuckets = new Bucket[NUM_ERROR_BUCKETS];
+      for (int i = 0; i < NUM_ERROR_BUCKETS; i++) {
+        errorBuckets[i] = new Bucket(NUM_SAMPLES_PER_ERROR_BUCKET);
+      }
+    }
+
+    private Bucket getLatencyBucket(long latencyNs) {
+      for (int i = 0; i < NUM_LATENCY_BUCKETS; i++) {
+        LatencyBucketBoundaries boundaries = LatencyBucketBoundaries.values()[i];
+        if (latencyNs >= boundaries.getLatencyLowerNs()
+            && latencyNs < boundaries.getLatencyUpperNs()) {
+          return latencyBuckets[i];
+        }
+      }
+      return latencyBuckets[NUM_LATENCY_BUCKETS];
+    }
+
+    private Bucket getErrorBucket(CanonicalCode code) {
+      return errorBuckets[code.value() - 1];
+    }
+
+    private void considerForSampling(SpanImpl span) {
+      Status status = span.getStatus();
+      Bucket bucket =
+          status.isOk()
+              ? getLatencyBucket(span.getLatencyNs())
+              : getErrorBucket(status.getCanonicalCode());
+      bucket.considerForSampling(span);
+    }
+
+    private Map<LatencyBucketBoundaries, Integer> getNumberOfLatencySampledSpans() {
```

Comment:
I think getNumbersOfLatencySampledSpans (or getLatencySampledSpanCounts) would be clearer, since it is a collection of numbers.  This also applies to the corresponding method in PerSpanNameSummary, though that would be a separate change.